### DISCUSSION TITLE
Build on Windows

### DISCUSSION THREAD

**Discussion by gabrofig** [open-quantum-safe/liboqs#1473] [Q&A]
I am a beginner and I am struggling to build liboqs in Windows. I was following the steps and first running:
 mkdir build && cd build
 cmake -GNinja ..

However, when running ninja, I keep having errors like bellow:
[382/387] Linking C executable tests\example_sig.exe
FAILED: tests/example_sig.exe
cmd.exe /C "cd . && D:\mingw64\bin\gcc.exe  -Wl,--stack,16777216 tests/CMakeFiles/example_sig.dir/example_sig.c.obj -o tests\example_sig.exe -Wl,--out-implib,tests\libexample_sig.dll.a -Wl,--major-image-version,0,--minor-image-version,0  lib/liboqs.a  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
d:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: lib/liboqs.a(hash_shakex4.c.obj):hash_shakex4.c:(.text$PQCLEAN_SPHINCSSHAKE128FSIMPLE_AVX2_prf_addrx4+0x163): undefined reference to `KeccakP1600times4_PermuteAll_24rounds'

Additionally, when trying the alternative using msbuild, and the suggested command in the documenation msbuild ALL_BUILD.vcxproj, I find the following message: MSBUILD : error MSB1009: file does not exist.

Is there anything I am missing installing or something I am not doing correctly?
I appreciate any feedback and apologies for the newbie question :)

**Comment by xvzcf** [comment]
I'm working on this.

**Comment by xvzcf** [comment]
We currently do not support SPHINCS in liboqs on Windows (see: [this](https://github.com/open-quantum-safe/liboqs/blob/main/appveyor_build.bat#L5), [this](https://github.com/open-quantum-safe/liboqs/blob/main/appveyor_build.bat#L10), and [this](https://github.com/open-quantum-safe/liboqs/blob/main/appveyor_build.bat#L18)).

I think this error might have to do with the fact that [OQS_ENABLE_SHA3_xkcp_low_avx2 is not enabled on Windows](https://github.com/open-quantum-safe/liboqs/blob/main/.CMake/alg_support.cmake#L74), but I don't further time at the moment to dig into it.

**Comment by xvzcf** [comment]
I've opened an issue here: https://github.com/open-quantum-safe/liboqs/issues/1476

**Comment by gabrofig** [comment]
Thank you, Goutam. I am trying to run KEM with the python wrapper, however, due the issue with the build in C, there is no folder C:\some\dir\liboqs\build\bin that I can add to my environment variable so the wrapper can import oqs package. Is there any workaround I could apply inorder to use the python wrapper?
Thank you very much, kind regards,

**Comment by baentsch** [comment]
@gabrofig Can you please try again with the latest "main"? https://github.com/open-quantum-safe/liboqs/pull/1478 landed and should have fixed the issue.

**Reply by baentsch** [reply]
@gabrofig Can you please confirm your issue is now fixed (marking the question answered)? Otherwise please provide further detail what still doesn't work.

**Reply by gabrofig** [reply]
Hi @baentsch, appologies for the delay! Yes, I can confirm that the issue looks solved. Super appreciated!
